opus/application/static_media/js/browse.js

```diff
@@ -749,17 +754,37 @@ var o_browse = {
             obsNum = (o_utils.floor((obsNum - 1)/galleryBoundingRect.x) *
                       galleryBoundingRect.x + 1);
 
+            // update scrollbarOffset in both infiniteScroll instances:
+            // Store scrollbar offset in scrollbarOffset, this will be used in the calculation
+            // in setScrollbarPosition and provide smooth scrolling when infiniteScroll load event
+            // is triggered.
+            let galleryTarget = $(`${tab} .op-thumbnail-container[data-obs="${obsNum}"]`);
+            let tableTarget = $(`${tab} .op-data-table tbody tr[data-obs='${tableCurrentObsNum}']`);
+            let galleryOffset = 0;
+            let tableOffset = 0;
+            if (galleryTarget.length && tableTarget.length) {
+                let galleryTargetTopPosition = galleryTarget.offset().top;
+                let galleryContainerTopPosition = $(`${tab} .gallery-contents .op-gallery-view`).offset().top;
+                galleryOffset = galleryTargetTopPosition - galleryContainerTopPosition;
+                let tableTargetTopPosition = tableTarget.offset().top;
+                let tableContainerTopPosition = $(`${tab} .op-data-table-view`).offset().top;
+                let tableHeaderHeight = $(`${tab} .op-data-table thead th`).outerHeight();
+                tableOffset = tableTargetTopPosition - tableContainerTopPosition - tableHeaderHeight;
+            }
+
             if (maxSliderVal >= obsNum) {
                 // "sliderObsNum" will be the startObs.
                 // "scrollbarObsNum" will be the obsNum at current scrollbar location.
                 // In table view, it's the top obsNum. In gallery view, it's one of obsNums in the top row.
                 $(`${tab} .op-gallery-view`).infiniteScroll({
                     "sliderObsNum": obsNum,
-                    "scrollbarObsNum": tableCurrentObsNum
+                    "scrollbarObsNum": tableCurrentObsNum,
```

Comment:
I'm not sure I understand this code, but why it scrollbarObsNum being set to tableCurrentObsNum here instead of obsNum? I would think we would want the gallery scrollbar to always be on a row boundary.